src/contexts/ApiProvider.tsx

```diff
@@ -1,80 +1,38 @@
-import { createContext, useEffect, useState } from 'react'
-import { WebSocketProvider } from './WebSocketProvider'
-import type { JWTPayload } from 'jose'
+import { createContext, useMemo } from 'react'
+import { HocuspocusProvider, HocuspocusProviderWebsocket } from '@hocuspocus/provider'
 
 interface ApiProviderProps {
   children: React.ReactNode
-  protocol: string
-  host: string
-  port: number
+  apiUrl: string
+  websocketUrl: string
 }
 
 export interface ApiProviderState {
-  api?: string
-  ws?: string
-  jwt?: JWTPayload
+  apiUrl: string
+  websocketUrl: string
```

Comment:
Would be nice to treat these as an `URL`. Might not be possible but it would simplify some of the handling, appending, validation...

Reply:
Agree. I'll convert it.